airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -855,3 +855,29 @@ def read_records(
         yield from self.filter_records_newer_than_state(stream_state, self.sort_output_asc(records))
         # add log message about the checkpoint value
         self.emit_checkpoint_message()
+
+
+class FullRefreshShopifyGraphQlBulkStream(ShopifyStream):
+    data_field = "graphql"
+    http_method = "POST"
+
+    query: DeliveryZoneList
+    response_field: str
+
+    def request_body_json(
+        self,
+        stream_state: Optional[Mapping[str, Any]],
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping[str, Any]]:
+        return {"query": self.query().get()}
+
+    @limiter.balance_rate_limit()
```

Comment:
Shouldn't we mention which type of API is used, like:
```python
@limiter.balance_rate_limit(api_type=ApiTypeEnum.graphql.value)
```
By default the `api_type=ApiTypeEnum.rest.value` is applied, which might be not correct in terms of the data structure to track the rate limit.